### PR TITLE
Early exit when no files are changed.

### PR DESCRIPTION
**Pull request by nathanhammond**
As a consequence of the `checkAll` function call returning `true` if the length of `changedFiles` is 0, this must early-exit in order to avoid labeling empty PRs.

Example labeling failure here: https://github.com/vercel/turbo/pull/2706

### CODE REVIEW

**Comment by nathanhammond** [comment]
The source of the error is here:
https://github.com/actions/labeler/blob/5c7539237e04b714afd8ad9b4aed733815b9fab4/src/labeler.ts#L202-L214

The loop on lines 205-210 does not execute and therefore the `checkAll` check returns `true`.

**Comment by nathanhammond** [comment]
Rebased and updated to match new code style. As all changes were formatting, the `dist` asset didn't change.

File: src/labeler.ts [hunk 1]
```diff
@@ -40,6 +40,23 @@ export async function run() {
 
     const labels: string[] = [];
     const labelsToRemove: string[] = [];
+
+    if (changedFiles.length === 0) {
```
Comment (dmitry-shibanov) [inline src/labeler.ts:44]: could you please change it to if (!changedFiles.length) {

File: src/labeler.ts [hunk 2]
```diff
@@ -40,6 +40,23 @@ export async function run() {
...
+      core.debug(`exiting early because pr #${prNumber} has no changed files.`);
+
+      if (syncLabels) {
+        for (let label of pullRequest.labels) {
+          if (label.name !== undefined) {
```
Comment (dmitry-shibanov) [inline src/labeler.ts:49]: I think we can change it to the same check if (label.name) {

Comment (nathanhammond) [reply src/labeler.ts:49]: Confirmed that name can't be blank at creation time. (returns 422)

`{"name":["Name can't be blank"],"message":"Name can't be blank"}`

**Comment by nathanhammond** [comment]
Updated and rebuilt.

**Comment by AndreiLobanovich** [comment]
Hello @nathanhammond! Could you please tell me, how do you create an empty PR? The link you provided as a labeler failure leads to a pull request. If you could provide the link with the run in which labeler failed - that would help me investigate the issue.

File: src/labeler.ts [hunk 3]
```diff
@@ -40,6 +40,23 @@ export async function run() {
 
     const labels: string[] = [];
     const labelsToRemove: string[] = [];
+
```
Comment (AndreiLobanovich) [inline src/labeler.ts:43]: Could you please move this right after the following line?
`const changedFiles: string[] = await getChangedFiles(client, prNumber);`

File: src/labeler.ts [hunk 4]
```diff
@@ -40,6 +40,23 @@ export async function run() {
...
+    if (!changedFiles.length) {
+      core.debug(`exiting early because pr #${prNumber} has no changed files.`);
```
Comment (AndreiLobanovich) [inline src/labeler.ts:45]: This should be a warning. Creating an empty pr should definitely be noticed by the user.

File: src/labeler.ts [hunk 5]
```diff
@@ -40,6 +40,23 @@ export async function run() {
...
+
+      if (syncLabels) {
```
Comment (AndreiLobanovich) [inline src/labeler.ts:47]: There is no need to remove labels, because early exit doesn't allow them to appear.

**Review by AndreiLobanovich — CHANGES_REQUESTED** [review]
Hello, @nathanhammond! 
Thanks for your contribution and attention to details! I managed to reproduce the incorrect behavior. Could you please make a few minor changes?

**Comment by AndreiLobanovich** [comment]
Hello, @nathanhammond!

Didn't hear from you for a while. Do you have some spare time to finish work on this PR? Is there anything I can help you with?

**Comment by AndreiLobanovich** [comment]
Hey, @nathanhammond, we miss you a lot. If you could find some spare time to finish work on this PR that would be great. 
Hope you are doing well.

**Review by AndreiLobanovich — APPROVED** [review]

**Review by IvanZosimov — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Comment by MaksimZhukov** [comment]
@nathanhammond thank you for the contribution!

**Pull request merged**